src/main/java/com/ecommerce/ECommerceApp/model/Order.java

```diff
@@ -91,7 +92,7 @@ public void setUserID(int userID) {
      *
      * @return the total amount of the order
      */
-    public float getTotalAmount() {
+    public BigDecimal getTotalAmount() {
```

Comment:
Replaced float with BigDecimal for monetary values to prevent rounding issues in financial calculations, ensuring precision and reliability. As Martin Fowler discusses in ["Technical Debt"](https://martinfowler.com/bliki/TechnicalDebt.html), addressing data accuracy early on reduces technical debt and avoids potential issues down the line.